packages/http-client/src/http-client.spec.ts

```diff
@@ -511,6 +511,100 @@ sap-client:001`);
       );
     });
 
+    it('rejectUnauthorized property of HttpsAgent should be set to true when isTrustingAllCertificates is false', async () => {
+      jest.spyOn(axios, 'request').mockResolvedValue({});
+      const destination: Destination = {
+        url: 'https://example.com',
+        isTrustingAllCertificates: true
+      };
+
+      await executeHttpRequest(destination, { method: 'get' });
+      const expectedJsonHttpsAgent = {
+        httpsAgent: expect.objectContaining({
+          options: expect.objectContaining({ rejectUnauthorized: false })
+        })
+      };
+      expect(axios.request).toHaveBeenCalledWith(
+        expect.objectContaining(expectedJsonHttpsAgent)
+      );
+    });
+
+    it('rejectUnauthorized property of HttpsAgent should be set to false when isTrustingAllCertificates is true', async () => {
+      jest.spyOn(axios, 'request').mockResolvedValue({});
+      const destination: Destination = {
+        url: 'https://example.com',
+        isTrustingAllCertificates: false
+      };
+      await executeHttpRequest(destination, { method: 'get' });
+      const expectedJsonHttpsAgent = {
+        httpsAgent: expect.objectContaining({
+          options: expect.objectContaining({ rejectUnauthorized: true })
+        })
+      };
+      expect(axios.request).toHaveBeenCalledWith(
+        expect.objectContaining(expectedJsonHttpsAgent)
+      );
+    });
+
+    it('request config contains headers without ETag value when there is no ETag config', async () => {
```

Comment:
I find this test more natural in the odata-request, because you can just call the `headers()` methods to get the headers and see that the `if-match` is not there.

Reply:
Makes sense, done.